crates/turbopack-ecmascript/src/references/esm/binding.rs

```diff
@@ -21,144 +21,118 @@ use crate::{
 
 #[turbo_tasks::value(shared)]
 #[derive(Hash, Debug)]
+pub struct EsmBindings {
+    pub bindings: Vec<EsmBinding>,
+}
+
+#[turbo_tasks::value_impl]
+impl EsmBindings {
+    #[turbo_tasks::function]
+    pub fn new(bindings: Vec<EsmBinding>) -> Vc<Self> {
+        EsmBindings { bindings }.cell()
+    }
+}
+
+#[derive(Hash, Clone, Debug, TaskInput)]
 pub struct EsmBinding {
     pub reference: Vc<EsmAssetReference>,
     pub export: Option<RcStr>,
     pub ast_path: Vc<AstPath>,
 }
 
-#[turbo_tasks::value_impl]
 impl EsmBinding {
-    #[turbo_tasks::function]
     pub fn new(
         reference: Vc<EsmAssetReference>,
         export: Option<RcStr>,
         ast_path: Vc<AstPath>,
-    ) -> Vc<Self> {
+    ) -> Self {
         EsmBinding {
             reference,
             export,
             ast_path,
         }
-        .cell()
     }
 }
 
 #[turbo_tasks::value_impl]
-impl CodeGenerateable for EsmBinding {
+impl CodeGenerateable for EsmBindings {
     #[turbo_tasks::function]
     async fn code_generation(
         self: Vc<Self>,
         _context: Vc<Box<dyn ChunkingContext>>,
     ) -> Result<Vc<CodeGeneration>> {
         let this = self.await?;
         let mut visitors = Vec::new();
-        let imported_module = this.reference.get_referenced_asset();
+        let bindings = this.bindings.clone();
 
-        fn make_expr(
-            imported_module: &str,
-            export: Option<&str>,
-            span: Span,
-            in_call: bool,
-        ) -> Expr {
-            let span = span.with_ctxt(SyntaxContext::empty());
-            if let Some(export) = export {
-                let mut expr = Expr::Member(MemberExpr {
-                    span,
-                    obj: Box::new(Expr::Ident(Ident::new(imported_module.into(), span))),
-                    prop: MemberProp::Computed(ComputedPropName {
-                        span,
-                        expr: Box::new(Expr::Lit(Lit::Str(Str {
-                            span,
-                            value: export.into(),
-                            raw: None,
-                        }))),
-                    }),
-                });
-                if in_call {
-                    expr = Expr::Seq(SeqExpr {
-                        exprs: vec![
-                            Box::new(Expr::Lit(Lit::Num(Number {
-                                span,
-                                value: 0.0,
-                                raw: None,
-                            }))),
-                            Box::new(expr),
-                        ],
-                        span,
-                    });
-                }
-                expr
-            } else {
-                Expr::Ident(Ident::new(imported_module.into(), span))
-            }
-        }
+        for item in bindings.into_iter() {
```

Comment:
Can we keep this as a function of `EsmBinding` (maybe not a turbo task)?
it already has too many levels of indentation